sources/packages/api/src/route-controllers/student-account-applications/models/student-account-application.dto.ts

```diff
@@ -0,0 +1,42 @@
+import { Allow, IsObject } from "class-validator";
+import { CreateStudentAPIInDTO } from "../../../route-controllers/student/models/student.dto";
+
+export class CreateStudentAccountApplicationAPIInDTO {
+  @IsObject()
+  submittedData: unknown;
+}
+
+export class StudentAccountApplicationSummaryAPIOutDTO {
+  id: number;
+  fullName: string;
+  submittedDate: Date;
+}
+
+export class StudentAccountApplicationAPIOutDTO {
+  id: number;
+  submittedData: unknown;
+}
+
+export class HasPendingStudentAccountApplicationAPIOutDTO {
+  hasPendingApplication: boolean;
+}
+
+/**
+ * Represents all the data needed to create a student account and user.
+ * Ministry will ensure that all the data is accurate and this payload will
+ * contain the source of truth that must be used to update the existing user
+ * and also create or update the student account as needed.
+ * !The data must be validate using a form.io dry run.
+ */
+export class StudentAccountApplicationApprovalAPIInDTO extends CreateStudentAPIInDTO {
```

Comment:
Seeing the name Approval in DTO made me feel like it is going to consist approval related data. Just a thought.

Reply:
All data here is part of the approval process. Is there a better suggestion for the name or the comments?

Reply:
Got it. Looked into the controller later. :😊